Improve editor layout

brittnylapierre opened the pull request:
Hey @SaschaAdler I'm not sure if there's an issue to go along with this but feel free to attach it!

SaschaAdler commented on services/admin/src/lib/components/shared/SideMenuContainer.svelte:
```diff
@@ -50,28 +75,62 @@
-  <div class="auto-align auto-align__a-stretch ">
-    <slot />
+<div bind:this={container} class="side-menu-container">
+  <div class:fixed-full-page={fullPage}>
```
Something here is confusing GitHub's syntax highlighting.

SaschaAdler commented on services/admin/src/lib/components/shared/SideMenuContainer.svelte:
```diff
@@ -1,10 +1,35 @@
+  $: pageNames = pageList ? pageList.map((el) => el["name"]) : [];
+  $: pageComponents = pageList ? pageList.map((el) => el["componentData"]) : [];
+  $: {
+    for (let key in instances) {
```
Purely stylistic thing, but if a variable is never going to change, I'd recommend using `const`. It's rare in my experience that `for (let key in keys)` is preferable to `for (const key in keys)`.

SaschaAdler commented on services/admin/src/lib/components/shared/SideMenuPageListButton.svelte:
```diff
@@ -4,13 +4,12 @@
-    background: var(--grey);
-    filter: brightness(1.1);
+    background: rgba(0, 0, 0, 0.06);
     padding: var(--perfect-fourth-7) var(--perfect-fourth-4);
```
I just noticed the musical ratio thing and I like it.

SaschaAdler commented on services/admin/src/routes/object/[prefix]/[noid].svelte:
```diff
@@ -1,65 +1,87 @@
   object = $objectStore;
   setDataModel(object);
 
+  let pageList: Array<SideMenuPageData> | undefined;
```
I'd instantiate `pageList` with `const pageList: Array<SideMenuPageData> = [];` and then push items into it. It'll help make the logic below less repetitive.

We might want to create a type guard for whatever I called the parent type of Collections and Manifests. (I wish I used traits instead. Maybe someday.)

SaschaAdler commented on services/admin/src/routes/object/[prefix]/[noid].svelte:
```diff
@@ -1,65 +1,87 @@
+  let pageList: Array<SideMenuPageData> | undefined;
+
   let rfdc: any; // Deep copies an object
   let model: AccessObject;
```
This is the unchanged copy, right? I'm not sure `model` is the best name for this; it's unclear what it's a model of.

SaschaAdler commented on services/admin/src/routes/object/[prefix]/[noid].svelte:
```diff
@@ -1,65 +1,87 @@
-  {/if}
+  <div class="editor">
+    <SideMenuContainer {pageList}>
+      <Toolbar slot="side-menu-header" title={object["slug"]}>
```
Not something to solve with this PR, necessarily, but we might want to extricate the Toolbar from the SideMenuContainer, for use on pages where we want it but no SideMenu.

brittnylapierre commented:
Thanks for your feedback Sascha! I'll make the required edits.

brittnylapierre commented:
@SaschaAdler I pushed fixes from the review! 

I couldn't make the pageList const otherwise it wouldn't draw the menu but I made it have the default value of []. 

I changed the variable model to objectModel and added a comment explaining it is where the changes to the object go before the user presses save. 

I added a bit more templating for the app colours (made the bg of the canvas viewer match what's on canadiana.ca now, so the app currently has more of a light theme.) Feel free to play around and change the defaults! It's pretty easy to make something like a dark mode now. If people want to clone this repo it should be easy for them to re-skin it with their own colours too. 

For the InfoEditor.svelte, I'm not sure if we should pass an object or if it would instead be better to have multiple of props (ex: slug, label, ...) that can be null, and show props that are instantiated only? If it matters, if you have a preference I can apply it to all components.
